docs/architecture/psa-migration/syms.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/sh
+#
+# Copyright The Mbed TLS Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Purpose
+#
+# Show symbols in the X.509 and TLS libraries that are defined in another
+# libmbedtlsXXX.a library. This is usually done to list Crypto dependencies.
```

Comment:
Do you think it would be useful to have a source-level analysis for that? It shouldn't be very hard with libclang (more work than this script, but not a lot more, and I have a few python scripts with some reusable bits of code, e.g. https://github.com/ARMmbed/mbedtls/pull/4800/files#diff-75f22d484b1a1dc724dded90a046d1c017dcec3f26e2e7cfa5d12e4baccd8b88).

Reply:
It could be nice but I'm not sure it's worth the effort, especially as I don't think we'll be doing this kind of analysis every other day. If I find myself using this script more often than I think and being frustrated that the results are too crude, I'll remember that better tools are not that hard to make.